### DISCUSSION TITLE
How to add a directory to GITHUB_PATH when creating a JavaScript action

### DISCUSSION THREAD

**Discussion by yusancky** [community/community#59391] [Actions] [answered]
### Select Topic Area

Question

### Body

How to add a directory to GITHUB_PATH when creating a JavaScript action?

**Comment by nirav-gajera** [comment, marked as answer]
1. Install the @actions/core package as a dependency in your JavaScript action. You can do this by including it in your package.json file or running npm install @actions/core in your action's directory.
2. In your JavaScript action file, import the core module from @actions/core:  `const core = require('@actions/core');
`
3. Use the core.addPath() method to add the directory path to the GITHUB_PATH:  
```
const directoryPath = '<path-to-directory>';
core.addPath(directoryPath);
```


-Replace <path-to-directory> with the actual path of the directory you want to add.

By adding the directory path using core.addPath(), it will be appended to the GITHUB_PATH environment variable and will be available for subsequent steps in your workflow to use.

**Reply by yusancky** [reply]
Thanks very much!
But please, how do I download a file from a GitHub Releases URL like `https://github.com/user/repo/releases/download/tag/file.zip`?

I have tried the following function but it doesn't seem to work well.

```javascript
async function downloadFromUrl(downloadUrl, fileName, token) {
  const response = await fetch(downloadUrl, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  if (response.ok) {
    const fileStream = fs.createWriteStream(fileName);
    await new Promise((resolve, reject) => {
      response.body.pipe(fileStream);
      fileStream.on("finish", () => {
        fileStream.close();
        resolve();
      });
      fileStream.on("error", (error) => {
        reject(error);
      });
    });
  } else {
    console.error("Failed to download file:", response.status, response.statusText);
  }
}
```

**Reply by nirav-gajera** [reply]
you can use the axios library

```
const axios = require('axios');
const fs = require('fs');

async function downloadFromUrl(downloadUrl, fileName) {
  const response = await axios({
    method: 'get',
    url: downloadUrl,
    responseType: 'stream',
  });

  const fileStream = fs.createWriteStream(fileName);
  response.data.pipe(fileStream);

  return new Promise((resolve, reject) => {
    fileStream.on('finish', () => {
      fileStream.close();
      resolve();
    });

    fileStream.on('error', (err) => {
      reject(err);
    });
  });
}

const downloadUrl = 'https://github.com/user/repo/releases/download/tag/file.zip';
const fileName = 'file.zip';

downloadFromUrl(downloadUrl, fileName)
  .then(() => {
    console.log('File downloaded successfully!');
  })
  .catch((error) => {
    console.error('Failed to download file:', error);
  });

```

You don't need to include the Authorization header with a bearer token for downloading public files from GitHub Releases. The code provided assumes you're downloading a public file. If you need authentication for private repositories, you may need to modify the code accordingly.

**Reply by yusancky** [reply]
I tried yours but it still failed. :sob:

```javascript
node:events:505
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open '/usr/local/typst-x86_64-unknown-linux-gnu.tar.gz'
Emitted 'error' event on ReadStream instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/usr/local/typst-x86_64-unknown-linux-gnu.tar.gz'
}
```

[Code - index.js (txt)](https://github.com/community/community/files/11893351/index.js.txt)

**Reply by nirav-gajera** [reply]
The error message suggests that the file /usr/local/typst-x86_64-unknown-linux-gnu.tar.gz does not exist on your system. The issue could be related to the file not being downloaded successfully or the incorrect file path being used.

1. Verify that the file is being downloaded correctly  `console.log('Archive downloaded:', fs.existsSync(archiveName));`
2. Check if the downloaded file has a different name or extension. Ensure that the assetName variable is correctly set based on your desired file name and extension.
3. Double-check the file path and permissions. Ensure that the file path `/usr/local/typst-x86_64-unknown-linux-gnu.tar.gz` is correct and that you have the necessary permissions to read and write to that location.
